tests/integ_tests/polling_tests.rs

```diff
@@ -88,3 +98,125 @@ async fn out_of_order_completion_doesnt_hang() {
 
     jh.await.unwrap();
 }
+
+#[tokio::test]
+async fn switching_worker_client_changes_poll() {
+    // Start two servers
+    info!("Starting servers");
+    let server_config = TemporalDevServerConfigBuilder::default()
+        .exe(default_cached_download())
+        // We need to lower the poll timeout so the poll call rolls over
+        .extra_args(vec![
+            "--dynamic-config-value".to_string(),
+            "matching.longPollExpirationInterval=\"1s\"".to_string(),
+        ])
+        .build()
+        .unwrap();
+    let mut server1 = server_config.start_server().await.unwrap();
+    let mut server2 = server_config.start_server().await.unwrap();
+
+    // Connect clients to both servers
+    info!("Connecting clients");
+    let client1 = ClientOptionsBuilder::default()
+        .identity("integ_tester".to_owned())
+        .target_url(Url::parse(&format!("http://{}", server1.target)).unwrap())
+        .client_name("temporal-core".to_owned())
+        .client_version("0.1.0".to_owned())
+        .build()
+        .unwrap()
+        .connect("default", None)
+        .await
+        .unwrap();
+    let client2 = ClientOptionsBuilder::default()
+        .identity("integ_tester".to_owned())
+        .target_url(Url::parse(&format!("http://{}", server2.target)).unwrap())
+        .client_name("temporal-core".to_owned())
+        .client_version("0.1.0".to_owned())
+        .build()
+        .unwrap()
+        .connect("default", None)
+        .await
+        .unwrap();
```

Comment:
Can save a lot of lines here by doing the common builder setup, cloning, then adding the specific target.